src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx

```diff
@@ -0,0 +1,50 @@
+"use client"
+
+import { useParams } from "next/navigation"
+import { PropsWithChildren, createContext, useContext, useEffect, useState } from "react"
+
+type CourseContextType = {
+    courseId: number
+    courseName: string
+}
+
+const CourseContext = createContext<CourseContextType>({
+  courseId: 0,
```

Comment:
default this to null and just don't send the request to fetch the course while courseId is null